src/Player.ts

```diff
@@ -82,18 +84,22 @@ export class Player implements ISerializable<SerializedPlayer> {
   public terraformRatingAtGenerationStart: number = 20;
 
   // Resources
-  public megaCredits: number = 0;
-  private megaCreditProduction: number = 0;
-  public steel: number = 0;
-  private steelProduction: number = 0;
-  public titanium: number = 0;
-  private titaniumProduction: number = 0;
-  public plants: number = 0;
-  private plantProduction: number = 0;
-  public energy: number = 0;
-  private energyProduction: number = 0;
-  public heat: number = 0;
-  private heatProduction: number = 0;
+  private units: Units = {
```

Comment:
I would leaverage a `Units.default()` and leverage it for `units` and `production`.

```
private units = Units.default();
private production = Units.default();
```